developer_manual/digging_deeper/profile.rst

```diff
@@ -0,0 +1,185 @@
+=======
+Profile
+=======
+
+The user profile presents the information of a user including their full name,
+profile picture, organisation, role, location, biography, headline, as well as
+actionable information that we call profile actions. These actions can include
+starting a `Nextcloud Talk <https://nextcloud.com/talk/>` chat with the user,
+sending an email, visiting their website, opening their Twitter profile,
+calling their phone number, scheduling an appointment, and more.
+
+App developers can integrate into profile and provide their own actions.
+
+Register a profile action
+-------------------------
+
+A profile action is represented by a class implementing the `OCP\\Profile\\ILinkAction`
+interface. This class is instantiated whenever the user profile is loaded.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace OCA\MyApp\Profile;
+
+    use OCP\Accounts\IAccountManager;
+    use OCP\IURLGenerator;
+    use OCP\IUser;
+    use OCP\IL10N;
+    use OCP\Profile\ILinkAction;
+
+    class MyProfileAction implements ILinkAction {
+
+        /** @var string */
+        private $targetUser;
+
+        /** @var IAccountManager */
+        private $accountManager;
+
+        /** @var IL10N */
+        private $l10n;
+
+        /** @var IUrlGenerator */
+        private $urlGenerator;
+
+        public function __construct(
+            IAccountManager $accountManager,
+            IL10N $l10n,
+            IURLGenerator $urlGenerator
+        ) {
+            $this->accountManager = $accountManager;
+            $this->l10n = $l10n;
+            $this->urlGenerator = $urlGenerator;
+        }
+
+        /**
+         * Preload the user specific value required by the action
+         *
+         * e.g. the email is loaded for the email action and the userId for the Talk action
+         *
+         * @since 23.0.0
+         */
+        public function preload(IUser $targetUser): void {
+            $this->targetUser = $targetUser;
+        }
+
+        /**
+         * Returns the app ID of the action
+         *
+         * e.g. 'spreed'
+         *
+         * @since 23.0.0
+         */
+        public function getAppId(): string {
+            return 'my_app_id';
```

Comment:
`Application::APP_ID` ?

Reply:
That would assume people have it defined in the application file and in a documentation I think it's better to not make cross assumptions

Reply:
On that it seems to be defined in the example code later on, so should be fine

Reply:
> That would assume people have it defined in the application file and in a documentation I think it's better to not make cross assumptions

Yeah, pretty much was making it more verbose as it would be trivial for app devs to make that tweak themselves if they so choose

For a first time app dev it would be clearer to keep this and they can do it DRY when they get the hang of it